Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -539,15 +539,90 @@ namespace AzToolsFramework
             }
         }
 
+        FixedSizeMessageBox::FixedSizeMessageBox(
+            const QString& title,
+            const QString& text,
+            const QString& informativeText,
+            const QString& detailedText,
+            QMessageBox::Icon icon,
+            QMessageBox::StandardButton standardButton,
+            QMessageBox::StandardButton defaultButton,
+            QWidget* parent)
+            : QMessageBox(parent)
+        {
+            setWindowTitle(title);
+            setText(text);
+            setInformativeText(informativeText);
+            setDetailedText(detailedText);
+            setIcon(icon);
+            setStandardButtons(standardButton);
+            setDefaultButton(defaultButton);
+        }
+
+        void FixedSizeMessageBox::SetSize(int width, int height)
+        {
+            QSpacerItem* horizontalSpacer = new QSpacerItem(width, height, QSizePolicy::Minimum, QSizePolicy::Expanding);
+            QGridLayout* layout = (QGridLayout*)this->layout();
+            layout->addItem(horizontalSpacer, layout->rowCount(), 0, 1, layout->columnCount());
+        }
+
         void AssetBrowserTreeView::RenameEntry()
         {
             auto entries = GetSelectedAssets(false); // you cannot rename product files.
-            // you may not rename products.
-            if (entries.size() == 1)
+
+            if (entries.size() != 1)
             {
-                edit(currentIndex());
+                return;
+            }
+            using namespace AzFramework::AssetSystem;
+            bool connectedToAssetProcessor = false;
+            AzFramework::AssetSystemRequestBus::BroadcastResult(
+                connectedToAssetProcessor, &AzFramework::AssetSystemRequestBus::Events::AssetProcessorIsReady);
+
+            if (connectedToAssetProcessor)
+            {
+                using namespace AZ::IO;
+                AssetBrowserEntry* item = entries[0];
+                Path fromPath = item->GetFullPath();
+                Path toPath(fromPath);
+                toPath.ReplaceExtension("renameFileTestExtension");
```

Comment:
What does the test extension do?

Reply:
It is used to allow the checkMove to decide what happens when a file is Moved within the same folder.